validator/src/main/java/com/google/daq/mqtt/registrar/LocalDevice.java

```diff
@@ -309,12 +307,17 @@ private DeviceCredential getDeviceCredential(String keyFile) throws IOException
   }
 
   private Set<String> keyFiles() {
-    if (!isDirectConnect()) {
+    if (!isDirectConnect() || !hasAuthType()) {
       return ImmutableSet.of();
     }
+    String authType = getAuthType();
+    Set<String> certFile = (authType.equals(ES_CERT_TYPE) || authType.equals(RSA_CERT_TYPE)) ? Set.of(certFile()) : null;
     Set<String> publicKeyFile = Set.of(publicKeyFile());
     Set<String> privateKeyFiles = privateKeyFiles();
-    return Sets.union(publicKeyFile, privateKeyFiles);
+    Set<String> rtn = (authType.equals(ES_CERT_TYPE) || authType.equals(RSA_CERT_TYPE)) ? 
+       Sets.union(Sets.union(publicKeyFile, certFile), privateKeyFiles) : 
```

Comment:
hanging indent should be 4 (I think it's 3 here?)

Reply:
yup, my bad, sorry